internal/config/validation.go

```diff
@@ -96,6 +97,53 @@ func DiscardNativeOnly(c ClusterResources) error {
 	return nil
 }
 
+// validateConfig is meant to validate all the inter-dependencies of a parsed configuration.
+// In this case, we ensure that bfd echo is not enabled on a pool.
+func validateConfig(cfg *Config) error {
+outer:
+	for _, p := range cfg.Pools.ByName {
+		containsV6 := false
+		for _, cidr := range p.CIDR {
+			if ipfamily.ForCIDR(cidr) == ipfamily.IPv6 {
+				containsV6 = true
+				break
+			}
+			if !containsV6 { // we only care about v6 advertisements
+				continue outer
+			}
+		}
+		for _, a := range p.BGPAdvertisements {
+			if len(a.Peers) == 0 {
```

Comment:
can you add a comment here please (why do we care about an advertisement that will not be advertised)?

Reply:
it will be advertised. len(peers) == 0 means all peers. Commenting